### PR TITLE
chore(EMS-3960): improve country mapping helper functions

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
Some country mapping functions (that consume CIS API data) were hard to read and follow.

There is likely to be some changes to this logic, it should be easy to do so.

## Resolution :heavy_check_mark:
- Update the following functions to have object structured params, instead of consuming and entire `mapped`/`country` object:
  - `canGetAQuoteOnline`
  - `canGetAQuoteByEmail`
  - `cannotGetAQuote`
- Create new `MapCisCountryParams` type.
- Rename all instanceds of `riskCategory` to `esraClassification`. In a future PR, the logic will use `esraClassification` and a new field, `countryRating`.

## Miscellaneous :heavy_plus_sign:
- Unit test improvements.
- Minor destructuring improvments.
- Remove unnecessary const exports.
- Add some logs and try/catch handlers to "generate quote" functions to improve debugging.

### CODE REVIEW

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
  * - Unlisted
- * @param {Boolean} shortTermCover: Boolean value of shortTermCover.
- * @param {String} riskCategory: Country risk category.
+ * @param {Boolean} shortTermCover: Short term cover flag.
+ * @param {String} esraClassification: ESRA classification.
  * @returns {Boolean}
  */
-export const canApplyForInsuranceOnline = (shortTermCover: boolean, riskCategory?: string) => {
-  if (riskCategory && shortTermCover) {
+const canApplyForInsuranceOnline = (shortTermCover: boolean, esraClassification?: string | null) => {
```
Comment (Zainzzkk) [inline src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts:16]: do we need ? if its going to be string | null? or is undefined expected too?

**Review by Zainzzkk — DISMISSED** [review]

**Review by Zainzzkk — COMMENTED** [review]
Do not merge actually, the tests have not been kicked off

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts (hunk 1 above)
Comment (ttbarnes) [reply src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts:16]: It is specifically going to be a string or null - i've made it required

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts [hunk 2]
```diff
@@ -9,12 +9,12 @@
...
  * - Unlisted
- * @param {Boolean} shortTermCover: Boolean value of shortTermCover.
- * @param {String} riskCategory: Country risk category.
+ * @param {Boolean} shortTermCover: Short term cover flag.
+ * @param {String} esraClassification: ESRA classification.
  * @returns {Boolean}
  */
-export const canApplyForInsuranceOnline = (shortTermCover: boolean, riskCategory?: string) => {
-  if (riskCategory && shortTermCover) {
+const canApplyForInsuranceOnline = (shortTermCover: boolean, esraClassification: string | null) => {
```
Comment (Zainzzkk) [inline src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts:16]: could this function just be `const canApplyForInsuranceOnline = (shortTermCover: boolean, esraClassification: string | null) => shortTermCover && esraClassification`?

File: src/ui/server/generate-quote/get-premium-rate/index.ts [hunk 3]
```diff
@@ -16,26 +16,34 @@ const PRICING_GRID_MAP = {
...
+    console.info('Getting premium rate');
 
-  const riskCategoryKey = PRICING_GRID_MAP.RISK_CATEGORY[riskCategory];
+    const policyTypeKey = PRICING_GRID_MAP.POLICY_TYPE[policyType];
 
-  const pricingGrid: any = PRICING_GRID as PricingGrid;
+    const esraClassificationKey = PRICING_GRID_MAP.RISK_CATEGORY[esraClassification];
 
-  const risk = pricingGrid[policyTypeKey][riskCategoryKey];
+    const pricingGrid: any = PRICING_GRID as PricingGrid;
```
Comment (Zainzzkk) [inline src/ui/server/generate-quote/get-premium-rate/index.ts:33]: should we  be using : any here?

Comment (ttbarnes) [reply src/ui/server/generate-quote/get-premium-rate/index.ts:33]: Good spot, updated.

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts (hunk 2 above)
Comment (ttbarnes) [reply src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts:16]: Updated.

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-get-a-quote-by-email/index.ts [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+ * @param {MapCisCountryParams}
+ * @param {Boolean} shortTermCover: Short term cover flag.
+ * @param {Boolean} nbiIssueAvailable: NBI flag.
+ * @param {String} esraClassification: ESRA classification.
  * @returns {Boolean}
  */
-export const canGetAQuoteByEmail = (country: MappedCisCountry) => {
-  if (country.riskCategory && country.shortTermCover && !country.nbiIssueAvailable) {
+const canGetAQuoteByEmail = ({ shortTermCover, nbiIssueAvailable, esraClassification }: MapCisCountryParams) => {
+  if (shortTermCover && !nbiIssueAvailable && esraClassification) {
```
Comment (Zainzzkk) [inline src/api/helpers/map-CIS-countries/map-CIS-country/can-get-a-quote-by-email/index.ts:13]: can we have a ticket to update this as well

File: src/api/helpers/map-CIS-countries/map-CIS-country/cannot-get-a-quote/index.ts [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
+ * @param {MapCisCountryParams}
+ * @param {Boolean} shortTermCover: Short term cover flag.
+ * @param {Boolean} nbiIssueAvailable: NBI flag.
+ * @param {String} esraClassification: ESRA classification.
  * @returns {Boolean}
  */
-export const cannotGetAQuote = (country: MappedCisCountry) => {
-  if (!country.riskCategory || (!country.shortTermCover && !country.nbiIssueAvailable)) {
+const cannotGetAQuote = ({ shortTermCover, nbiIssueAvailable, esraClassification }: MapCisCountryParams) => {
+  if (!esraClassification || (!shortTermCover && !nbiIssueAvailable)) {
```
Comment (Zainzzkk) [inline src/api/helpers/map-CIS-countries/map-CIS-country/cannot-get-a-quote/index.ts:13]: can add this to the above ticket to update this as well

File: src/api/helpers/map-CIS-countries/map-CIS-country/map-NBI-issue-available/index.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ const { CIS } = EXTERNAL_API_DEFINITIONS;
  * @param {String} NBI flag
  * @returns {Boolean}
  */
-export const mapNbiIssueAvailable = (str: string): boolean => {
+const mapNbiIssueAvailable = (str: string): boolean => {
   if (str === CIS.NBI_ISSUE_AVAILABLE.YES) {
```
Comment (Zainzzkk) [inline src/api/helpers/map-CIS-countries/map-CIS-country/map-NBI-issue-available/index.ts:12]: can add this to the above ticket to update this as well

**Review by Zainzzkk — APPROVED** [review]

File: src/api/helpers/map-CIS-countries/map-CIS-country/map-NBI-issue-available/index.ts (hunk 6 above)
Comment (ttbarnes) [reply src/api/helpers/map-CIS-countries/map-CIS-country/map-NBI-issue-available/index.ts:12]: RE these comments, i'd rather leave as they are. The mapping logic will be changed, most likely throughout, in EMS-3842.

File: src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts [hunk 7]
```diff
@@ -9,16 +9,10 @@
...
  * @returns {Boolean}
  */
-export const canApplyForInsuranceOnline = (shortTermCover: boolean, riskCategory?: string) => {
-  if (riskCategory && shortTermCover) {
-    return true;
-  }
-
-  return false;
-};
+const canApplyForInsuranceOnline = (shortTermCover: boolean, esraClassification: string | null) => shortTermCover && esraClassification;
```
Comment (ttbarnes) [inline src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts:16]: @Zainzzkk Had to revert this change, the one liner will not work as it can return string | null | boolean. It needs to be a boolean at all times.

Comment (Zainzzkk) [reply src/api/helpers/map-CIS-countries/map-CIS-country/can-apply-for-insurance-online/index.ts]: thats fine, will approve.  Otherwise we could have wrapped it in Boolean like `=> Boolean(shortTermCover && esraClassification);`

**Review by Zainzzkk — APPROVED** [review]
